### PR TITLE
Add new version 2018-04-02 for Microsoft.StorageSync Resource Provider

### PR DESCRIPTION
**Pull request by ankushbindlish2**
Add new version 2018-04-02 for Microsoft.StorageSync Resource Provider

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by ankushbindlish2** [comment]
Preview Version : https://github.com/Azure/azure-rest-api-specs/pull/3211

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/examples/RegisteredServers_Create.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    "serverId": "\"3635ca8b-5cc8-4a5c-bd43-c2de5ad8dc64\"",
+                    "storageSyncServiceUid": "\"4aa14534-1c61-483b-b6a6-9630a76f109a\"",
+                    "lastWorkflowId": "storageSyncServices/CV_FileStore_F1D485AA/workflows/4eecfbcf-9537-4b61-8fee-aaa3ace11c44",
+                    "lastOperationName": "ICreateRegisteredServerWorkflow",
+                    "friendlyName": "afscv-2304-139",
+                    "managementEndpointUri": "\"https://kailanitest99.one.microsoft.com:443/\"",
+                    "discoveryEndpointUri": "\"https://kailanitest99.one.microsoft.com:443\"",
+                    "resourceLocation": "westus",
+                    "serviceLocation": "westus"
+                },
```
Comment (jianghaolu) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/examples/RegisteredServers_Create.json]: Invalid JSON - remove extra comma

**Comment by jianghaolu** [comment]
@annatisch Seems like you reviewed the preview version - do you plan to review this one?

**Review by anuchandy — CHANGES_REQUESTED** [review]
@ankushbindlish2 could you please follow the instructions I mentioned in the email i.e. splitting this PR into two commits, review will be fast when we can see the diff.

**Comment by ankushbindlish2** [comment]
@anuchandy  : Did you mean like this ? https://github.com/Azure/azure-rest-api-specs/pull/3238

**Pull request closed without merging**